benchexec/tools/aise.py

```diff
@@ -0,0 +1,89 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.result as result
+import benchexec.tools.template
+import benchexec.model
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for AISE.
+    """
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("aise", subdir="bin")
+
+    def program_files(self, executable):
+        return self._program_files_from_executable(
+            executable, self.REQUIRED_PATHS, parent_dir=True
+        )
+
+    def cmdline(self, executable, options, task, rlimits):
+        return [executable] + options + list(task.input_files_or_identifier)
+
+    def name(self):
+        return "AISE"
+
+    def version(self, executable):
+        return "AISE 1.0.0"
```

Comment:
Hard-coding a version number in this module is not a good idea. If people run a newer version of your tool with BenchExec they will get a wrong version number in the output, which could be very confusing.

The best would be to add a parameter like `--version` to your tool and read and return the actual version number here. You can call the utility method `self._version_from_tool` to do this easily.

If this is not possible, please remove the method completely. (But please note that SV-COMP and Test-Comp AFAIK require a proper version number.)

Also, the version number should not duplicate the tool name.

Reply:
Ok, I added a parameter to return version.